cpp/models/abm/parameters.h

```diff
@@ -434,6 +435,36 @@ using MigrationParameters =
     ParameterSet<LockdownDate, SocialEventRate, BasicShoppingRate, WorkRatio, SchoolRatio, GotoWorkTimeMinimum,
                  GotoWorkTimeMaximum, GotoSchoolTimeMinimum, GotoSchoolTimeMaximum>;
 
+/**
+ * @brief Parameters that control the migration between Location%s.
+ */
+using AllParameters =
+    ParameterSet<IncubationPeriod, SusceptibleToExposedByCarrier, SusceptibleToExposedByInfected, CarrierToInfected,
+                 CarrierToRecovered, InfectedToRecovered, InfectedToSevere, SevereToCritical, SevereToRecovered,
+                 CriticalToDead, CriticalToRecovered, RecoveredToSusceptible, DetectInfection, MaskProtection,
+                 LockdownDate, SocialEventRate, BasicShoppingRate, WorkRatio, SchoolRatio, GotoWorkTimeMinimum,
+                 GotoWorkTimeMaximum, GotoSchoolTimeMinimum, GotoSchoolTimeMaximum>;
+
+class SimulationParameters : public AllParameters
+{
+public:
+    SimulationParameters(AgeGroup num_agegroups)
+        : AllParameters(num_agegroups)
+        , m_num_groups{num_agegroups}
+    {
+    }
+
+    AgeGroup get_num_groups() const
+    {
+        return m_num_groups;
+    }
+
+    static const int32_t DEFAULT_NUM_AGE_GROUPS = 6;
```

Comment:
Does this need to be all caps. As we don't use that elsewhere it looks confusing.